test/jdk/java/util/Currency/ValidateISO4217.java

```diff
@@ -168,12 +168,17 @@ private static void processColumns(StringTokenizer tokens, String country) throw
                 if (format == null) {
                     createDateFormat();
                 }
-                // If the cut-over already passed, test the changed data too
+                // If the cut-over already passed, use the new curency for ISO4217Codes
                 if (format.parse(tokens.nextToken()).getTime() < System.currentTimeMillis()) {
                     currency = tokens.nextToken();
                     numeric = tokens.nextToken();
                     minorUnit = tokens.nextToken();
                     testCurrencies.add(Currency.getInstance(currency));
+                } else {
+                    // Still need to add the future currency to testCurrencies
+                    // without updating ISO4217Codes
+                    String futureCurr = tokens.nextToken();
+                    testCurrencies.add(Currency.getInstance(futureCurr));
```

Comment:
Updated the Currency build process to disallow any Currencies that are future Currencies. This prevents the future currency, "XCG" from leaking out into `Currency.getAvailableCurrencies()`. (This was only exposed now, since the previous future Currencies were already ones expected to be found in `Currency.getAvailableCurrencies()`. E.g. Amendment 174 where HRK was replaced by EUR.

Reply:
IIUC, this should have happened before, as a currency update like "XXA" -> "XXB" on a future date is pretty usual. I think this issue was simply not discovered by any users till now.

Reply:
I think this wasn't discovered/hasn't failed the test before because the amendments with future date changes most of the time were implemented after the date already occurred, so there was no cut-over, simply a new currency directly replacing the old. In the case where the cut-over hadn't happened, like 174 where EUR replaced HRK, EUR is already expected to be returned by `Currency.getAvailableCurrencies()` so there was no issue.

I claimed that this particular amendment exposed it, because if this case had happened before with an amendment update, it would have caused validateISO4217.java to fail (as it does with this case). Since that test checks if the set returned by the method does not contain any future currencies.

Reply:
Yes, it is rare but happened before. The test had failed with the cut-over date in the past. Possibly at that time, getAvailableCurrencies test may not have been there so it wasn't obvious when the modification was made.